packages/file-asset-apis/src/s3/s3-helpers.ts

```diff
@@ -162,11 +162,14 @@ export async function doesBucketExist(
     } catch (err) {
         const { httpStatusCode } = (err as S3ClientResponse.S3ErrorExceptions).$metadata ?? {};
 
-        if (httpStatusCode === 404) {
-            return false;
-        }
         if (httpStatusCode === 403) {
             throw new TSError(`User does not have access to bucket "${params.Bucket}"`, { statusCode: 403 });
+        // In the case of a 4** status code, return false
+        } else if (
+            Number(httpStatusCode) >= 400 &&
+            Number(httpStatusCode) <= 500
```

Comment:
```suggestion
            Number(httpStatusCode) < 500
```

We probably don't want to include 500.